File: app/src/main/java/org/metabrainz/mobile/data/Resource.java
package org.metabrainz.mobile.data;

public class Resource<T> {
    private final Status status;
    private final T data;

    public Resource(Status status, T data) {
        this.status = status;
        this.data = data;
    }

    public Status getStatus() {
        return status;
    }

    public T getData() {
        return data;
    }

    public enum Status {
        LOADING, FAILED, SUCCESS
    }
}

File: app/src/main/java/org/metabrainz/android/util/Resource.kt
package org.metabrainz.android.util

class Resource<T>(val status: Status, val data: T?) {

    enum class Status {
        LOADING, FAILED, SUCCESS
    }

    companion object {
        fun <S> failure(): Resource<S> {
            return Resource(Status.FAILED, null)
        }
        fun <S> loading(): Resource<S> {
            return Resource(Status.LOADING, null)
        }
    }
}